platform/engineering/design-docs/2020-09-27-broken-links.md

```diff
@@ -0,0 +1,122 @@
+
+### Background
+Broken links discovered during the Front End content build currently prevent content releases.  Notifications are sent to Slack #CMS-Team.  This pulls CMS team members away from their planned work, to triage the situation.  This includes:
+1. Logging into Jenkins to find the report.
+2. Determining if the broken link comes from the CMS.
+3. Reporting it to the editor or team that may be responsible.
+4. Notify all editors in #CMS-Support and via email that content is NOT being released. (Email is required because most of the editors in the CMS are not on Slack.)
+5. Notifying in #CMS-Support and via email that the issue has been resolved and content is now flowing as expected.
+The effort is time consuming and unproductive.
+While people are scrambling to investigate and notify affected editors, content is not going out.  Emergency alerts _(active shooter, hurricane closures, blizzard closures...)_ do not go out.  Email notifications from GovDelivery which are dependent on a FE content release, do not go out.  Timely releases of new initiatives, do not go out.
+
+The CMS team built the [Node Link Report](https://www.drupal.org/project/node_link_report) module (open sourced module) to detect and warn of broken links upon node save. Editors now have immediate feedback about potential broken links before publishing. This greatly reduced the number of broken link related content release stoppages.  We have done as much as we can on that front at the CMS layer.
+
+
+### Guiding Principles
+1. Veterans should not get 404 errors. Content on VA.gov should not have broken links.
+2. As the CMS scales from ~20 editors and daily content releases, up to ~500 editors and continuous content releases, we can not allow a broken link to prevent content releases.  Knowing about a storm closure or active shooter [saves more souls](https://www.informationweek.com/government/dont-get-side-tracked-by-flashy-metrics-success-is-in-the-people/d/d-id/1331791) than not getting a 404 when following a link.
+3. Developers currently waste far too much time chasing down broken links.
+
+
+### High Level Design
+Currently the Front End link checker creates a CSV file and exposes that to Jenkins.  The CSV contains the location of the broken link, the url it is pointing to.  I propose publishing that file to the website as open data (perhaps located at /data/vagov-broken-links.csv or as json).  Currently the Front End build process notifies the CMS when a content release has succeeded by pinging the CMS  The CMS could act on that ping as a trigger and read the broken link data.  It could process it immediately and take the following actions:
```

Comment:
I'm not certain, but we might have to find a different namespace other than `/data`. 

Currently, https://staging.va.gov/data redirects to https://www.data.va.gov/.